tutorials/kb_guided_llm_ner/ner_llm_kb_correct.py

```diff
@@ -0,0 +1,215 @@
+    """
+    for eg in stream:
+        if eg["has_ent_errors"] or eg["has_label_errors"]:
+            yield eg
+
+
+def add_ner_annotations(
+    stream: StreamType,
+    nlp: Language,
+    labels: Iterable[str],
+    batch_size: int = DEFAULT_BATCH_SIZE,
+) -> StreamType:
+    """Add a 'spans' key to each example, with entities predicted by LLM
+    and 'kb_spans' key, with entities matched by KB.
+
+    stream (iterable): The stream of annotation tasks.
+    nlp (Language): The spaCy pipeline.
+    labels (Iterable[str]): The entity labels to annotate.
+    batch_size (int): The batch size for the LLM.
+    YIELDS (dict): The annotation examples.
+    """
+    texts = ((eg["text"], eg) for eg in stream)
+    for doc, eg in nlp.pipe(texts, as_tuples=True, batch_size=batch_size):
+        task = copy.deepcopy(eg)
+        llm_spans = []
+        kb_spans = []
+        for ent in doc.ents:
+            if labels and ent.label_ not in labels:
+                continue
+            llm_spans.append(
+                {
+                    "token_start": ent.start,
+                    "token_end": ent.end - 1,
+                    "start": ent.start_char,
+                    "end": ent.end_char,
+                    "text": ent.text,
+                    "label": ent.label_,
+                }
+            )
+        task["spans"] = llm_spans
+
+        def parse_types(raw_results: Dict) -> List[str]:
+            return [
+                db_type.split(":")[1]
+                for db_type in raw_results["@types"].split(",")
+                if db_type.startswith("DBpedia")
+            ]
+
+        for ent in doc.spans["dbpedia_spotlight"]:
+            kb_spans.append(
+                {
+                    "token_start": ent.start,
+                    "token_end": ent.end - 1,
+                    "start": ent.start_char,
+                    "end": ent.end_char,
+                    "text": ent.text,
+                    "types": parse_types(ent._.dbpedia_raw_result),
+                }
+            )
+        task["kb_spans"] = kb_spans
+        task["llm"] = doc.user_data["llm_io"]["llm"]
+        yield task
+
+
+@recipe(
+    # fmt: off
+    "ner.llm.kb.correct",
+    dataset=("Dataset to save answers to", "positional", None, str),
+    config_path=("Path to the spacy-llm config file", "positional", None, str),
+    source=("Data to annotate (file path or '-' to read from standard input)", "positional", None, str),
+    labels=("Comma-separated label(s) to annotate or text file with one label per line", "positional", None, split_string),
+    loader=("Loader (guessed from file extension if not set)", "option", "lo", str),
+    segment=("Split articles into sentences", "flag", "S", bool),
+    skip_validated=("Skip examples that are validated by KB", "flag", "V", bool),
+    # fmt: on
+)
+def llm_kb_correct_ner(
+    dataset: str,
+    config_path: Path,
+    source: Union[str, Iterable[dict]],
+    labels: List[str],
```

Comment:
Might be nice to throw an error if the labels in the config file don't appear here. 

Or maybe ... does the user really need to pass labels here? Could we just re-use the labels in the `spacy.cfg`? I can understand the need to maybe select a subset, but then labels should be an optional arg.